src/theory/sets/theory_sets_rels.cpp

```diff
@@ -125,44 +125,61 @@ void TheorySetsRels::check(Theory::Effort level)
 
     TERM_IT t_it = d_terms_cache.begin();
     while( t_it != d_terms_cache.end() ) {
```

Comment:
As we discussed internally, this loop can be merged with the previous loop in this [line](https://github.com/cvc5/cvc5/blob/master/src/theory/sets/theory_sets_rels.cpp#L76). However, this can be done in a future PR. 
Also the relational solver needs to be refactored to avoid unnecessary cashing for membership constraints which are already cashed in SolverState.